### PR TITLE
BBOX camera support

### PR DESCRIPTION
**Pull request by ianthetechie**
* Adds support for a bounding box camera mode (it will revert to centered after the transition completes, but this isn't a problem IMO)
* Upgrades dependencies

### CODE REVIEW

File: compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraToMapLibre.kt [hunk 1]
```diff
@@ -21,6 +22,18 @@ internal fun MapViewCamera.toCameraPosition(): CameraPosition {
...
     }
+    is CameraState.BoundingBox -> {
+      return map.getCameraForLatLngBounds(
+          this.state.bounds,
+          intArrayOf(
+              this.padding.top.toInt(),
+              // TODO: Internally the C++ lib uses left/right, not start/end
+              this.padding.start.toInt(),
+              this.padding.bottom.toInt(),
+              this.padding.end.toInt()),
```
Comment (ianthetechie) [inline compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraToMapLibre.kt:33]: I'm not sure if we need to do something here or not... the C++ layer of course has no concept of localized layouts. We sorta localized our insets. Should the padding behavior depend on this?

File: compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraZoom.kt [hunk 2]
```diff
@@ -17,6 +17,7 @@ import kotlin.math.roundToInt
 fun MapViewCamera.getZoom(): Double {
   return when (this.state) {
     is CameraState.Centered -> this.state.zoom
+    is CameraState.BoundingBox -> TODO("This is gonna be a problem...")
```
Comment (ianthetechie) [inline compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraZoom.kt:20]: This file raises an interesting problem... While the camera will never remain in the bbox state for long (as currently implemented), we may need to make these methods failable (optional return values / silently ignoring updates). The whole point of a bbox abstraction is that you don't know the zoom and several other parameters until you compute it with a concrete map object.

File: compose/src/main/java/com/maplibre/compose/runtime/nodes/MapCameraNode.kt [hunk 3]
```diff
@@ -103,7 +104,9 @@ private class CameraTransitionListener(
 }
 
 private fun cameraUpdate(map: MapboxMap, camera: MapViewCamera) {
-  val cameraUpdate = CameraUpdateFactory.newCameraPosition(camera.toCameraPosition())
+  val cameraPosition = camera.toCameraPosition(map) ?: return
```
Comment (ianthetechie) [inline compose/src/main/java/com/maplibre/compose/runtime/nodes/MapCameraNode.kt:107]: What's this? Well, it could technically fail (per the C++ API), so we may need to bail. I assume this could only happen for an invalid bbox.

File: compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraToMapLibre.kt [hunk 4]
```diff
@@ -21,6 +22,18 @@ internal fun MapViewCamera.toCameraPosition(): CameraPosition {
           .tilt(this.state.pitch)
           .bearing(this.state.direction)
     }
+    is CameraState.BoundingBox -> {
+      return map.getCameraForLatLngBounds(
```
Comment (ianthetechie) [inline compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraToMapLibre.kt:26]: Easy to miss, but noting here that we explicitly return early. The `CameraPosition` builder does not have any way to construct a bounding box position. You have to directly ask the map for this, so this is a different flow. The padding logic at the end is also skipped because this method is supposed to take care of the padding.

**Review by ianthetechie — COMMENTED** [review]
Some comments to guide the discussion.

File: compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraToMapLibre.kt (hunk 1 above)
Comment (ianthetechie) [reply compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraToMapLibre.kt:33]: One other thing I noticed is that padding does not seem to have any effect on Android and I'm not sure why. I have looked at the C++ impl: https://github.com/maplibre/maplibre-native/blob/e6e8cb9d5145034e3856b7c0e183fb93f54fdda3/platform/android/MapboxGLAndroidSDK/src/cpp/native_map_view.cpp#L424. It looks like the C++ is using doubles, but the JNI accepts a list of ints? Not really sure what's up there or if it's a problem 🤷‍♂️

Comment (ianthetechie) [reply compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraToMapLibre.kt]: Update: I figured out the padding. The PR is now correct (ignoring the question of start/end) and I've added a test that asserts the findings.

File: compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraToMapLibre.kt (hunk 4 above)
Comment (Archdoog) [reply compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraToMapLibre.kt:26]: Makes sense. We should throw this in there as a comment 👍

File: compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraZoom.kt (hunk 2 above)
Comment (Archdoog) [reply compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraZoom.kt:20]: I would argue you should still be able to get zoom. On set zoom I could see it as reasonable behavior where a user:

1. Sets the camera to a bounding box.
2. Taps zoom in/zoom out and expects the camera to move just like centered.

In this case, we'd probably have to do the following for each step:

1. Move the camera to the bounding box state.
2. Convert the bbox camera to a centered state at the current "view port", then zoom in. 

Basically this lets us abandon the BBox restriction, while still obeying the user's goal of zooming in after centering on the BBox.

This same general logic flow seems to make sense for all of these methods. The only trick is we have to swap the camera to centered to allow the behavior.

**Review by Archdoog — CHANGES_REQUESTED** [review]
Have some notes about the setZoom, etc. They're mostly guided by what would probably be the user's behavior and expectation(s).

Otherwise, tried this out and it's looking nice!

File: compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraZoom.kt (hunk 2 above)
Comment (ianthetechie) [reply compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraZoom.kt:20]: Any state except the tracking ones will end up back at centered on both iOS and Android given how our observability works. This happens automatically and immediately after the animation completes.

The only point at which this could fail is in the interim if the user ninja taps the explicit zoom button to try to increment or something. At this point in time, the camera state value that we store cannot possibly have a zoom in the case of the bbox state. This would behave unexpectedly in the first case too by the way, but it would not crash the app.

Hence I think we probably just need to make this method return a `Double?`.

Comment (Archdoog) [reply compose/src/main/java/com/maplibre/compose/camera/extensions/MapViewCameraZoom.kt:20]: Yep. That'll work. I was overthinking per the fact that the MapView technically knows zoom even when we're not using it. But since we can reconstruct a bounding box repeatably, a `Double?` solves the problem I was thinking about.

**Review by Archdoog — APPROVED** [review]
